scheduler/state_machine/state_machine_utils.py

```diff
@@ -778,29 +778,28 @@ def dialog_to_be_completed(user_id: int) -> Optional[UserInterventionState]:
             UserInterventionState.completed.is_(False),
             UserInterventionState.last_time.isnot(None)
         )
-        .first()
+        .all()
     )
 
-    session.expunge(uncompleted)
+    for dialog in uncompleted:
+        session.expunge(dialog)

```

Comment:
This is tricky, because with lists, expunge doesn't work properly. You have to remap the values like:         

```suggestion
    result = [UserInterventionState(id=task.id,
                                        users_nicedayuid=task.users_nicedayuid,
                                        intervention_phase_id=task.intervention_phase_id,
                                        intervention_component_id=task.intervention_component_id,
                                        completed=task.completed,
                                        last_time=task.last_time,
                                        last_part=task.last_part,
                                        next_planned_date=task.next_planned_date,
                                        task_uuid=task.task_uuid) for task in uncompleted]
```